src/utils/breakdown_validation.py

```diff
@@ -0,0 +1,87 @@
+"""Function to validate the breakdown totals from the imputation."""
+import os
+import logging
+import pandas as pd
+from pandas import DataFrame as pandasDF
+
+
+BreakdownValidationLogger = logging.getLogger(__name__)
+
+def breakdown_validation(df: pd.DataFrame) -> pd.DataFrame:
+    bool_dict = {}
+    msg = ""
+
+    df['check1'] = df['222'] + df['223'] != df['203'] #
+    df['check2'] = df['202'] + df['203'] != df['204'] #
+    df['check3'] = df['205'] + df['206'] + df['207'] != df['204'] #
+    df['check4'] = df['221'] > df['209'] #
+    df['check5'] = df['219'] + df['220'] + df['209'] != df['210'] #
+    df['check6'] = df['204'] + df['210'] != df['211'] #
+    df['check7'] = (df['212'] + df['214'] + df['216'] + df['242'] + df['250'] + df['243'] + df['244'] + df['245'] + df['246'] != df['218']) & (df['211'] != df['218']) #
+    df['check8'] = df['225'] + df['226'] + df['227'] + df['228'] + df['229'] + df['237'] != df['218'] #
+    df['check9'] = df['302'] + df['303'] + df['304'] != df['305'] #
+    df['check10'] = df['501'] + df['503'] + df['505'] != df['507'] #
+    df['check11'] = df['502'] + df['504'] + df['506'] != df['508'] #
+    df['check12'] = df['405'] + df['407'] + df['409'] != df['411'] #
+    df['check13'] = df['406'] + df['408'] + df['410'] != df['412'] #
+
+    false_df = df[df.isin([True]).any(axis=1)]
+    
+    if len(false_df) > 0 :
+        msg += "There are issues with the logic of the columns.\n "
+
+    for index, row in false_df.iterrows():
```

Comment:
What happens if there are no issues, and it's an empty dataframe - does this need to be in the if statement?